addons/udes_stock/models/stock_inventory.py

```diff
@@ -84,12 +104,41 @@ def _is_adjusting_reserved(self):
                 return True
         return False
 
+    def _has_theoretical_quantity_changed(self):
+        """Check if any theoretical quantity in the location has changed.
+        """
+        self.ensure_one()
+        for line in self.line_ids:
+            current_qty = sum(line._get_quants().mapped("quantity"))
+            if (line.theoretical_qty or current_qty) and current_qty != line.theoretical_qty:
+                return True
+        return False
+
+    def _compute_conflicting_theoretical_qty_message(self):
+        """Generate a html message to be displayed to the user when attempting to
+        validate a inventory adjustment when the theoretical quantity has changed
+        """
+        self.ensure_one()
+        message = "<ul>"
+        for line in self.line_ids:
+            current_qty = sum(line._get_quants().mapped("quantity"))
+            if (line.theoretical_qty or current_qty) and current_qty != line.theoretical_qty:
+                message += "<li>" + line.location_id.name + ": " + line.product_id.name
+                message += " " + line.prod_lot_id.name if line.prod_lot_id else ""
+                message += " " + line.package_id.name if line.package_id else ""
+                message += "; Old Theoretical Quantity: " + str(line.theoretical_qty)
+                message += ", <b>New Theoretical Quantity: " + str(current_qty) + "</b>"
+                message += "</li>"
```

Comment:
I would suggest to use .format(), example 
message += "; Old Theoretical Quantity: " + str(line.theoretical_qty) == >
message += "; Old Theoretical Quantity: {}".format(line.theoretical_qty)

Reply:
Might be worth also putting this logic into a `get_line_info` or similar function? That way might be a bit easier to deal with overrides for customers like @apatrick88  mentioned.

Reply:
Done both of these suggestions. The identifying information is now in a separate function that can be overwritten when needed

Reply:
Also added translation of those strings at Peter's suggestion